### PR TITLE
[SPARK-7060][SQL] Add alias function to python dataframe

### PR DESCRIPTION
**Pull request by yhuai**
This pr tries to provide a way to let python users workaround https://issues.apache.org/jira/browse/SPARK-6231.

### CODE REVIEW

**Comment by yhuai** [comment]
jenkins test this please

**Comment by rxin** [comment]
Jenkins, retest this please.

**Comment by yhuai** [comment]
Jenkins, retest this please.

File: python/pyspark/sql/dataframe.py [hunk 1]
```diff
@@ -453,6 +453,19 @@ def columns(self):
         return [f.name for f in self.schema.fields]
 
     @ignore_unicode_prefix
+    def alias(self, alias):
+        """Returns a new :class:`DataFrame` with an alias set.
+
+        >>> df_alias1 = df.alias("df_alias1")
+        >>> df_alias2 = df.alias("df_alias2")
+        >>> joined_df = df_alias1.join(df_alias2, df_alias1.name == df_alias2.name, 'inner')
```
Comment (marmbrus) [inline python/pyspark/sql/dataframe.py]: `joined_df = df_alias1.join(df_alias2, col("df_alias1.name") == col("df_alias2.name"), 'inner')`

**Comment by rxin** [comment]
Thanks. Meging in master.

**Pull request closed without merging**